db.go

```diff
@@ -250,15 +263,27 @@ type DB struct {
 	mu                 sync.RWMutex
 	models             map[reflect.Type]*Model
 	mappings           map[reflect.Type]fieldMap
+	IgnoreUnmappedCols bool
 }
 
 // NewDB creates a new DB from an sql.DB.
 func NewDB(db *sql.DB) *DB {
+	return newDBInternal(db, false)
+}
+
+// NewDBIgnoringUnmappedCols creates a new DB from an sql.DB and ignores unknown
```

Comment:
This change is great and the tests are great, but the name `NewDBIgnoringUnmappedCols` is a bit of a mouthful. Do you really need a new method here. Perhaps you can take the same approach as `DB.AllowStringQueries` and just allow the user to set the `DB.IgnoreUnmappedCols` field after creating the DB.